Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Select.hs

```diff
@@ -177,6 +197,19 @@ select2 personId estimateId req@DSelectReq {..} = do
     QP.updateDefaultPaymentMethodId paymentMethodId personId -- Make payment method as default payment method for customer
   when ((searchRequest.validTill) < now) $
     throwError SearchRequestExpired
+  when (maybe False Trip.isDeliveryTrip (DEstimate.tripCategory estimate)) $ do
+    when (isNothing deliveryDetails) $ throwError $ InvalidRequest "Delivery details not found for trip category Delivery"
+    let validDeliveryDetails = fromJust deliveryDetails
+    let senderLocationId = searchRequest.fromLocation.id
+    when (isNothing searchRequest.toLocation) $ throwError $ InvalidRequest "Receiver location not found for trip category Delivery"
```

Comment:
Same could be handled with fromMaybeM instead of validating and using fromJust

Reply:
✅